pkg/logs/internal/decoder/auto_multiline_handler.go

```diff
@@ -25,28 +26,56 @@ type AutoMultilineHandler struct {
 }
 
 // NewAutoMultilineHandler creates a new auto multiline handler.
-func NewAutoMultilineHandler(outputFn func(m *message.Message), maxContentSize int, flushTimeout time.Duration, tailerInfo *status.InfoRegistry) *AutoMultilineHandler {
+func NewAutoMultilineHandler(outputFn func(m *message.Message), maxContentSize int, flushTimeout time.Duration, tailerInfo *status.InfoRegistry, sourceSettings *config.SourceAutoMultiLineOptions, sourceSamples []*config.AutoMultilineSample) *AutoMultilineHandler {
 
 	// Order is important
 	heuristics := []automultilinedetection.Heuristic{}
 
-	heuristics = append(heuristics, automultilinedetection.NewTokenizer(pkgconfigsetup.Datadog().GetInt("logs_config.auto_multi_line.tokenizer_max_input_bytes")))
-	heuristics = append(heuristics, automultilinedetection.NewUserSamples(pkgconfigsetup.Datadog()))
+	tokenizerMaxInputBytes := pkgconfigsetup.Datadog().GetInt("logs_config.auto_multi_line.tokenizer_max_input_bytes")
+	if sourceSettings != nil && sourceSettings.TokenizerMaxInputBytes != nil {
```

Comment:
Can we move `sourceHasSettings := sourceSettings != nil` above this, and also use it here? Alternatively, we could expect this function to require a non-nil `sourceSettings` to avoid the `nil` checks.

Reply:
> how could we avoid the nil checks?
We would still ned to do the `sourceSettings.EnableJSONAggregation != nil` checks. 
However you could avoid the `sourceHasSettings := sourceSettings != nil` and `if sourceHasSettings &&` if we set the expectation that this struct is required. 

For example - `tailerInfo` is required. If you pass a `nil` tailer info, we will panic. 
One could argue that the `nil` check is safer, and in this case it won't actually save us that much code, so it may not be worth doing here (especially since I think `source.Config().AutoMultilineSamples` can in fact be nil). But I just wanted to call it out as an alternative.

Reply:
Ah okay makes sense